boards/arm/96b_neonkey/96b_neonkey.dts

```diff
@@ -39,7 +39,7 @@
 
 	gpio_keys {
 		compatible = "gpio-keys";
-		user_button: button@0 {
+		user_button: button_0 {
```

Comment:
I have no issues with this, but isn't "@0" part of the DTC spec?

Reply:
It is, but for reasons I don't understand using an '@' means that this becomes a unit-name. And when you have a unit-name you need a 'reg' property. To avoid a dummy 'reg' property we omit the '@' and avoid the 'reg' requirement.

Reply:
@SebastianBoe In that case, it is better to avoid '_' also and just use `button1`.

Reply:
Maybe, but I would prefer to follow prior art:

https://github.com/zephyrproject-rtos/zephyr/commit/7b0ce85242414c9391198d953c07e3858b1a11d0

https://github.com/zephyrproject-rtos/zephyr/pull/10335/commits/e5965143eab6d2c42a61c39f6d847d83b0d582aa